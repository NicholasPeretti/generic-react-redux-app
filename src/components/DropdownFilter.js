import React from 'react'
import PropTypes from 'prop-types'
import Style from '../style'
import styled from 'styled-components'
import Dropdown from './Dropdown'
import Text from './Text'

const FilterContainer = styled.div`
  display: flex;
  flex-direction: column;
`

const Label = styled(Text)`
  margin-bottom: ${Style.spacing.padding1};
`

var DropdownFilter = ({ label, value, items, onChange }) => (
  <FilterContainer>
    <Label>{label}</Label>
    <Dropdown
      value={value}
      items={items}
      onChange={onChange}
    />
  </FilterContainer>
)

DropdownFilter.propTypes = {
  label: PropTypes.string.isRequired,
  value: PropTypes.string,
  items: PropTypes.arrayOf(
    PropTypes.string
  ),
  onChange: PropTypes.func
}

export default DropdownFilter
