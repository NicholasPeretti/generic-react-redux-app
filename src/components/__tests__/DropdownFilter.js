import React from 'react'
import { shallow, mount } from 'enzyme'
import PresentationalTest from '../../utils/tests/presentational'
import DropdownFilter from '../DropdownFilter'

const props = {
  label: 'Test label',
  items: ['a', 'b'],
  onChange: jest.fn()
}

describe('DropdownFilter', () => {
  PresentationalTest(DropdownFilter, props)

  test('Should match snapshot', () => {
    const wrapper = shallow(<DropdownFilter {...props} />)
    expect(wrapper.html()).toMatchSnapshot()
  })

  test('Should call `onChange` when the dropdown changes', () => {
    const onChange = jest.fn()
    const wrapper = mount(<DropdownFilter {...props} onChange={onChange}/>)

    wrapper.find('button').simulate('click')
    wrapper.find('li').last().simulate('click')

    expect(onChange).toHaveBeenCalled()
  })

  test('Should call initialFetch if passed as prop', () => {
    const initialFetch = jest.fn()
    const wrapper = mount(<DropdownFilter {...props} initialFetch={initialFetch}/>)

    wrapper.find('button').simulate('click')
    wrapper.find('li').last().simulate('click')

    expect(initialFetch).toHaveBeenCalled()
  })
})
