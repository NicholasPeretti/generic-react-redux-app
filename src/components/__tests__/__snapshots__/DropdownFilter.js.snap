// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DropdownFilter Should match snapshot 1`] = `"<div class=\\"sc-bZQynM AMncx\\"><span class=\\"sc-bdVaJa sc-gzVnrw hlocEI\\">Test label</span><div class=\\"sc-bwzfXH gYNwqV\\"><button class=\\"sc-htpNat kYidVU\\"><span class=\\"sc-bdVaJa bkmfbd\\">a</span><span class=\\"sc-bxivhb kAOnLj\\"><span>â–¼</span></span></button></div></div>"`;
